cpu/samd21/periph/i2c.c

```diff
@@ -508,7 +508,7 @@ static inline void _stop(SercomI2cm *dev)
     /* Stop command */
     dev->CTRLB.reg |= SERCOM_I2CM_CTRLB_CMD(3);
     /* Wait for bus to be idle again */
-    while(dev->STATUS.reg & SERCOM_I2CM_STATUS_BUSSTATE(1)) {}
+    while((dev->STATUS.reg & SERCOM_I2CM_STATUS_BUSSTATE_Msk) != SERCOM_I2CM_STATUS_BUSSTATE(1)) {}
```

Comment:
This is the only one that actually caused me problems. We want the busstate to be 0b01, whereas the existing code waits until the LSB is clear. In my case it was a race and fell through because it was Owner (0b10), and the next call failed because it was still in Owner, but this could actually deadlock if the busstate somehow became idle (0b01) before this loop began because it would never leave that state (presumably the caller holds the i2c mutex).